### PR TITLE
ci: Infrastructure to run PyConsole tests in GitHub with Fluent docker image

### PR DESCRIPTION
**Pull request by mkundu1**
A new CI has been added to run Python test journals in the Fluent docker container. During the test run, the PyFluent library within the docker container is replaced with the library built from the current PyFluent branch. The primary goal is to test the impact of PyFluent changes in Fluent's PyConsole before adding those changes in Fluent.

### CODE REVIEW

File: .ci/fluent_test_runner.py [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+    image_name = "ghcr.io/ansys/pyfluent:v25.1.0"
+    container = docker_client.containers.run(
+        image=image_name,
+        volumes=[
+            f"{src_pyfluent_dir}:{dst_pyfluent_dir}",
+            f"{src_test_dir}:{dst_test_dir}",
+        ],
+        working_dir=dst_test_dir,
+        environment={"ANSYSLMD_LICENSE_FILE": os.environ["ANSYSLMD_LICENSE_FILE"]},
+        command=f"3ddp -gu -i {journal_file.name}",
```
Comment (mkundu1) [inline .ci/fluent_test_runner.py:54]: TODO: Each test will specify the launch arguments

File: .github/workflows/test-fluent-tests.yml [hunk 2]
```diff
@@ -0,0 +1,79 @@
+name: Test Fluent tests with PyFluent changes
+
+on:
+  pull_request:
```
Comment (mkundu1) [inline .github/workflows/test-fluent-tests.yml:4]: I'll change this to "on push - main branch" before merge

File: .ci/fluent_test_runner.py [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+        command=f"3ddp {launcher_args} -gu -py -i {journal_file.name}",
+        detach=True,
+        stdout=True,
+        stderr=True,
+    )
+    while True:
+        container.reload()
+        if container.status == "exited":
+            break
+        stderr = container.logs(stdout=False, stderr=True)
```
Comment (mkundu1) [inline .ci/fluent_test_runner.py:68]: TODO: process logs incrementally - will do in a later PR

**Comment by mkundu1** [comment]
Latest run - https://github.com/ansys/pyfluent/actions/runs/11168068525/job/31045656727#step:13:22

**Comment by hpohekar** [comment]
> A new CI has been added to run Python test journals in the Fluent docker container. During the test run, the PyFluent library within the docker container is replaced with the library built from the current PyFluent branch. The primary goal is to test the impact of PyFluent changes in Fluent's PyConsole before adding those changes in Fluent.

@mkundu1 Looks awesome ! Thank you.

**Review by hpohekar — APPROVED** [review]

**Review by seanpearsonuk — APPROVED** [review]

**Review by prmukherj — APPROVED** [review]

**Pull request merged**